moped-editor/src/components/GridTable/Filters.js

```diff
@@ -435,10 +461,11 @@ const Filters = ({
   useEffect(() => {
     if (Object.keys(filterParameters).length === 0) {
       setFilters(filterParameters);
+      handleClearFilters();
       // Add an empty filter so the user doesn't have to click the 'add filter' button
       generateEmptyFilter();
     }
-  }, [filterParameters, setFilters, generateEmptyFilter]);
+  }, [filterParameters, setFilters, generateEmptyFilter, handleClearFilters]);
```

Comment:
I thought about refactoring this so that the initial filter state has no filter objects in it instead of a blank placeholder with uuid, but I think that is too big and this works for what we are looking to do. It just makes it a little difficult to rely on the count to show/hide or anything else. I ended up creating some booleans vars that describe what is going on more and removing some checks that weren't actually doing anything but idk.

**Initial state**
```
{
    "id": "79f79105-4718-4df4-a614-fb665ec12b52",
    ...emptyFilterDetailsAsPlaceholder
}
```

**State after completing the first filter**
```
{
    "id": "79f79105-4718-4df4-a614-fb665ec12b52",
    ...completeFilterDetails
}
```

Reply:
yeah, that is tempting for sure. i can see how the initial filter creates lots of headaches. i will capture this tech debt here: https://github.com/cityofaustin/atd-data-tech/issues/10538